examples/redshift/v1beta1/endpointaccess.yaml

```diff
@@ -0,0 +1,120 @@
+apiVersion: redshift.aws.upbound.io/v1beta1
+kind: EndpointAccess
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: example
+  name: example-endpointaccess
+spec:
+  forProvider:
+    clusterIdentifierSelector:
+      matchLabels:
+        testing.upbound.io/example-name: example-endpointaccess
+    region: us-west-1
+    subnetGroupNameSelector:
+      matchLabels:
+        testing.upbound.io/example-name: example-endpointaccess
+---
+apiVersion: redshift.aws.upbound.io/v1beta1
+kind: Cluster
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: example-endpointaccess
+  name: example-endpointaccess-c
+spec:
+  forProvider:
+    clusterType: single-node
+    databaseName: mydb
+    masterPasswordSecretRef:
+      key: example-key
+      name: cluster-secret
+      namespace: upbound-system
+    masterUsername: exampleuser
+    nodeType: ra3.xlplus
+    region: us-west-1
+    skipFinalSnapshot: true
+---
+apiVersion: redshift.aws.upbound.io/v1beta1
+kind: SubnetGroup
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: example-endpointaccess
+  name: example-endpointaccess-sg
+spec:
+  forProvider:
+    region: us-west-1
+    subnetIdRefs:
+    - name: foo
+    - name: bar
+    tags:
+      environment: Production
+---
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: bar
+  name: bar
+spec:
+  forProvider:
+    availabilityZone: us-west-1b
+    cidrBlock: 10.1.2.0/24
+    region: us-west-1
+    tags:
+      Name: tf-dbsubnet-test-2
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: foo
+---
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: foo
+  name: foo
+spec:
+  forProvider:
+    availabilityZone: us-west-1a
+    cidrBlock: 10.1.1.0/24
+    region: us-west-1
+    tags:
+      Name: tf-dbsubnet-test-1
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: foo
+---
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: VPC
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+  labels:
+    testing.upbound.io/example-name: foo
+  name: foo
+spec:
+  forProvider:
+    cidrBlock: 10.1.0.0/16
+    region: us-west-1
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    meta.upbound.io/example-id: redshift/v1beta1/endpointaccess
+    uptest.upbound.io/pre-delete-hook: testhooks/delete-cluster.sh
```

Comment:
Actually, I did not test it without testhook, I opened this PR to use it in the guide I prepared about [adding a new resource](https://github.com/crossplane/upjet/pull/405). So I used the existing and working [`Cluster.redshift`](https://github.com/crossplane-contrib/provider-upjet-aws/blob/main/examples/redshift/v1beta1/cluster.yaml)resource. I do not intend to handle this in this PR to avoid confusion for community members who will add new resources.